src/vector/init.ts

```diff
@@ -40,31 +40,21 @@ export function preparePlatform() {
     }
 }
 
-export async function loadConfig(): Promise<{configError?: Error, configSyntaxError: boolean}> {
+export async function loadConfig(): Promise<Error | void> {
     const platform = PlatformPeg.get();
 
     let configJson;
-    let configError;
-    let configSyntaxError = false;
     try {
         configJson = await platform.getConfig();
     } catch (e) {
-        configError = e;
-
-        if (e && e.err && e.err instanceof SyntaxError) {
-            console.error("SyntaxError loading config:", e);
-            configSyntaxError = true;
-            configJson = {}; // to prevent errors between here and loading CSS for the error box
-        }
+        return e;
```

Comment:
I guess you return rather than throw the error here to receive the error without a try/catch in `app.js:190`. Returning errors from a promise is rather non-idiomatic though. Can I suggest to do it the more straight-forward way and throw and catch?

Reply:
I mean that's exactly what this code did previously as per Travis' doing except it was returning an error and a boolean

Reply:
it just gives a better progression to the app load, means first paint is sooner.

Reply:
Ah, yeah, missed that it was already there. Not sure how returning rather than throwing will allow first paint sooner but not too big of a deal, I'm fine either way.

Reply:
commented that in the wrong place